05week/spaceTravelToMars.js

```diff
@@ -1,51 +1,110 @@
-'use strict';
+"use strict";
 
-let assert = require('assert');
+let assert = require("assert");
 
 let jobTypes = {
-  pilot: 'MAV',
-  mechanic: 'Repair Ship',
-  commander: 'Main Ship',
-  programmer: 'Any Ship!'
+  pilot: "MAV",
+  mechanic: "Repair Ship",
+  commander: "Main Ship",
+  programmer: "Any Ship!",
 };
 
-// Your code here
+// class represents a crew memeber
+// crewmemeber should have a name, job, and specialSkill
+class CrewMember {
+  constructor(name, job, specialSkill) {
+    this.name = name;
+    this.job = job;
+    this.specialSkill = specialSkill;
+  }
+
+  //this method should ask THIS crewmember to the ship being passed in
+  //an entire ship instance is passed in, not just the name
+  //the entire crewmember is added to the ship's array of crew
+  enterShip(ship) {
+    ship.crew.push(this);
+  }
+}
+
+// class represents a ship
+// ship has a name, a type, an ability, crew array
+// ship constructor should set an array called crew to be any empty array
+class Ship {
+  constructor(name, type, ability) {
+    this.name = name;
+    this.type = type;
+    this.ability = ability;
+    this.crew = [];
+  }
+
+  // ship should have a method called missionStatement()
+  // this method should return "Can't perform mission yet." if it has no crew
+  // this method should return the ship's ability, if there is atleast 1 crew member
+  missionStatement() {
+    if (this.crew.length <= 0) {
+      return "Can't perform mission yet.";
+    } else if (this.crew.length >= 1) {
+      return `${this.ability}`;
```

Comment:
I do not see the check that looks for a match between the ship's type and crew member's job